src/raw.rs

```diff
@@ -27,6 +27,17 @@ pub struct RawTerminal<W: Write> {
 }
 
 #[cfg(not(target_os = "redox"))]
+impl<W> RawTerminal<W>
```

Comment:
I actually ran those lines through rustfmt, so i was a little confused, but it turns out [you're right](https://doc.rust-lang.org/book/traits.html#where-clause) and that's the style the Book uses for the examples.

Apparently rustfmt was actually bailing on formatting the file because some docstrings were over 99 characters long, time to file a bug report.